File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/ChatError.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class ChatError(
  val kind: String? = null,
  val name: String,
  val message: String,
  val retryAfter: String? = null,
  val limit: Long? = null,
  val userMessage: String? = null,
  val retryAfterDate: Date? = null,
  val retryAfterDateString: String? = null,
  val retryMessage: String? = null,
  val feature: String? = null,
  val upgradeIsAvailable: Boolean? = null,
  val isChatErrorGuard: IsChatErrorGuardEnum, // Oneof: isChatErrorGuard
) {

  enum class IsChatErrorGuardEnum {
    @SerializedName("isChatErrorGuard") IsChatErrorGuard,
  }
}

File: agent/bindings/csharp/ChatError.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class ChatError
  {
    [JsonProperty(PropertyName = "kind")]
    public string Kind { get; set; }
    [JsonProperty(PropertyName = "name")]
    public string Name { get; set; }
    [JsonProperty(PropertyName = "message")]
    public string Message { get; set; }
    [JsonProperty(PropertyName = "retryAfter")]
    public string RetryAfter { get; set; }
    [JsonProperty(PropertyName = "limit")]
    public int Limit { get; set; }
    [JsonProperty(PropertyName = "userMessage")]
    public string UserMessage { get; set; }
    [JsonProperty(PropertyName = "retryAfterDate")]
    public Date RetryAfterDate { get; set; }
    [JsonProperty(PropertyName = "retryAfterDateString")]
    public string RetryAfterDateString { get; set; }
    [JsonProperty(PropertyName = "retryMessage")]
    public string RetryMessage { get; set; }
    [JsonProperty(PropertyName = "feature")]
    public string Feature { get; set; }
    [JsonProperty(PropertyName = "upgradeIsAvailable")]
    public bool UpgradeIsAvailable { get; set; }
    [JsonProperty(PropertyName = "isChatErrorGuard")]
    public IsChatErrorGuardEnum IsChatErrorGuard { get; set; } // Oneof: isChatErrorGuard

    public enum IsChatErrorGuardEnum
    {
      [EnumMember(Value = "isChatErrorGuard")]
      IsChatErrorGuard,
    }
  }
}
